w10h02/test/pgdp/teams/UnitTest.java

```diff
@@ -47,4 +47,104 @@ public void optimalLineupAllAttackers() throws NoSuchFieldException, IllegalAcce
         assertEquals(0, defenders.size());
         assertEquals(0, supporters.size());
     }
+    
+    
+    @Test
+    @DisplayName("Soll funktionieren wenn nur defenders null elemente hat")
+    public void optimalLineUpZeroInMiddle() throws NoSuchFieldException, IllegalAccessException {
+        var lineup = Lineup.computeOptimalLineup(Set.of(jasmine, yassine), 1, 0, 1);
+
+        var attackers = getField(lineup, "attackers");
+        var defenders = getField(lineup, "defenders");
+        var supporters = getField(lineup, "supporters");
+
+        assertEquals(1, attackers.size());
+        assertEquals(0, defenders.size());
+        assertEquals(1, supporters.size());
+
+        assertTrue(attackers.contains(yassine));
+        assertTrue(supporters.contains(jasmine));
+        assertEquals(0, defenders.size());
+    }
+
+    @Test
+    @DisplayName("Soll funktionieren wenn attackers + defenders + supporters < players.size()")
+    public void optimalLineUpSubsetsDontSumUpToTotalSetSize() throws NoSuchFieldException, IllegalAccessException {
+        var lineup = Lineup.computeOptimalLineup(Set.of(jasmine, yassine, eve, anatoly, anton, cedric, faid, jakob, georg),
+                2, 2, 3);
+
+        var attackers = getField(lineup, "attackers");
+        var defenders = getField(lineup, "defenders");
+        var supporters = getField(lineup, "supporters");
+
+        assertEquals(2, attackers.size());
+        assertEquals(2, defenders.size());
+        assertEquals(3, supporters.size());
+
+        assertEquals(Set.of(eve, georg), attackers);
+        assertEquals(Set.of(jasmine, anton), defenders);
+        assertEquals(Set.of(jakob, yassine, faid), supporters);
+    }
+
+    @Test
+    @DisplayName("Soll funktionieren wenn defenders \"groß\" ist")
+    public void optimalLineUpLargeSubsetSize() throws NoSuchFieldException, IllegalAccessException {
+        var lineup = Lineup.computeOptimalLineup(Set.of(
+                jasmine, yassine, eve, anatoly, anton, cedric, faid, jakob, georg, hanna, jakov,
+                        felix, jan, shrek), 0, 10, 0);
+
+        var attackers = getField(lineup, "attackers");
+        var defenders = getField(lineup, "defenders");
+        var supporters = getField(lineup, "supporters");
+
+        assertEquals(0, attackers.size());
+        assertEquals(10, defenders.size());
+        assertEquals(0, supporters.size());
+
+        assertEquals(Set.of(eve), attackers);
+        assertEquals(Set.of(jan), defenders);
+        assertEquals(Set.of(jakob), supporters);
```

Comment:
Warum ist hier jeweils ein Pinguin in jedem Set?
`attackers` und `supporters` müsste doch leer sein und für defenders würde bei meiner Implementierung
```java
assertEquals(Set.of(jakob, jan, felix, eve, shrek, jakov, anatoly, hanna, jasmine, anton), defenders);
```
passen